### PR TITLE
[MRG+1] [BF] Fixes SpaceNet crashes when n_alphas=1 specified

### PR DESCRIPTION
**Pull request by dohmatob**
Fixes issue #852

### CODE REVIEW

**Comment by AlexandreAbraham** [comment]
Please give a meaningful name to your PR in the future.

File: nilearn/decoding/space_net.py [hunk 1]
```diff
@@ -823,8 +826,8 @@ def fit(self, X, y):
             y = y[:, 0]
 
         # generate fold indices
-        if (None in [alphas, l1_ratios] and self.n_alphas > 1) or min(
-                len(l1_ratios), len(alphas)) > 1:
+        if (None in [alphas, l1_ratios] and self.n_alphas > 1
+        ) or (not alphas is None and min(len(l1_ratios), len(alphas)) > 1):
```
Comment (AlexandreAbraham) [inline nilearn/decoding/space_net.py]: Is that PEP8 valid? Looks ugly.

Comment (dohmatob) [inline nilearn/decoding/space_net.py]: Simplified.

File: nilearn/decoding/space_net.py [hunk 2]
```diff
@@ -823,8 +826,9 @@ def fit(self, X, y):
             y = y[:, 0]
 
         # generate fold indices
-        if (None in [alphas, l1_ratios] and self.n_alphas > 1) or min(
-                len(l1_ratios), len(alphas)) > 1:
+        case1 = None in [alphas, l1_ratios] and self.n_alphas > 1
+        case2 = not alphas is None and min(len(l1_ratios), len(alphas)) > 1
+        if case1 or case2:
```
Comment (GaelVaroquaux) [inline nilearn/decoding/space_net.py]: I am not sure that I prefer the new version. But nevermind.

**Comment by GaelVaroquaux** [comment]
The AppVeyor failure is spurious: the build simply got cancelled (I am not sure why).

And the codacy failure is simply that codacy agrees with me on the rewriting of the test, I believe.

**Comment by GaelVaroquaux** [comment]
Hence, +1 for merge anyhow.

**Comment by dohmatob** [comment]
Ok, probably due to the superfluous for loop.  Fixed.

**Comment by bthirion** [comment]
LGTM too.

**Comment by GaelVaroquaux** [comment]
@dohmatob : I am not sure why, but this branch now has conflicts with master, and should be rebased.

File: nilearn/decoding/tests/test_space_net.py [hunk 3]
```diff
@@ -333,3 +333,11 @@ def test_crop_mask_empty_mask():
     assert_raises_regex(ValueError, "Empty mask:.", _crop_mask, np.array([]))
     assert_raises_regex(ValueError, "Empty mask:", _crop_mask,
                         np.zeros((2, 2, 2)))
+
+
+def test_space_net_one_alpha_no_crash():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, mask = to_niimgs(X, [2, 2, 2])
```
Comment (banilo) [inline nilearn/decoding/tests/test_space_net.py]: comment to mark as regression test?

**Comment by lesteve** [comment]
This needs a rebase on master.

File: nilearn/decoding/space_net.py [hunk 4]
```diff
@@ -823,8 +826,9 @@ def fit(self, X, y):
             y = y[:, 0]
 
         # generate fold indices
-        if (None in [alphas, l1_ratios] and self.n_alphas > 1) or min(
-                len(l1_ratios), len(alphas)) > 1:
+        case1 = (None in [alphas, l1_ratios]) and self.n_alphas > 1
+        case2 = (not alphas is None) and min(len(l1_ratios), len(alphas)) > 1
```
Comment (lesteve) [inline nilearn/decoding/space_net.py]: Nitpick: `alphas is not None and ...`

File: nilearn/decoding/space_net.py [hunk 5]
```diff
@@ -458,8 +458,11 @@ def path_scores(solver, X, y, mask, alphas, l1_ratios, train, test,
             all_test_scores.append(this_test_scores)
     else:
         if alphas is None:
-            raise RuntimeError
-        alphas_ = alphas
+            alphas_ = _space_net_alpha_grid(
```
Comment (lesteve) [inline nilearn/decoding/space_net.py]: Not sure why you did this in this PR, but you changed the meaning of `alphas=None` can you add a test for this as well?

Comment (dohmatob) [inline nilearn/decoding/space_net.py]: That line is directly related to the bug being addressed. Test added.

**Comment by lesteve** [comment]
Just in case you missed it: Travis fails for Python 2.6, not immediately obvious why.

**Comment by dohmatob** [comment]
I missed it. Thanks for spotting. Looks like this is related to the way fold indices are generated in `sklearn=0.13`.

**Comment by dohmatob** [comment]
This should be fixed now. Issue was that `y` was binarized before the fold generation code.

**Comment by GaelVaroquaux** [comment]
LGTM. Merging.

**Pull request merged**